stellargraph/utils/random.py

```diff
@@ -0,0 +1,65 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+import random
+import tensorflow as tf
+
+
+_sg_seed = None
+
+
+def get_seed(seed):
+    """
+    Convenience function to use the global seed by default if the provided seed is None.
+
+    Args:
+        seed (int, optional): seed value
+
+    Returns:
+        seed if not None, otherwise the global seed
+
+    """
+    if seed is None:
+        return _sg_seed
+    else:
+        return seed
+
+
+def set_seed(seed, set_np_seed=True, set_tf_seed=True, set_random_seed=True):
```

Comment:
I think the `_seed` in these is somewhat redundant, and potentially even the `set_` (given the function is already called `set_seed`. Also, this function is rarely called, so I think it's probably ok to spell the names out to avoid any ambiguity:

```suggestion
def set_seed(seed, set_numpy=True, set_tensorflow=True, set_random=True):
```